src/std/net/socket/server-test.ss

```diff
@@ -0,0 +1,84 @@
+;;; -*- Gerbil -*-
+;;; (C) vyzo at hackzen.org
+;;; Socket server tests
+
+(import :gerbil/gambit/threads
+        :std/misc/threads
+        :std/net/socket
+        :std/net/socket/base
+        :std/os/socket
+        :std/error
+        :std/sugar
+        :std/test)
+(export socket-server-test)
+
+(def socket-server-test
+  (test-suite "test :std/net/socket"
+
+    (def server-address "localhost:4242")
+
+    (def (echo sock)
+      (def buf (make-u8vector 256))
+      (try
+       (let lp ()
+         (def len (ssocket-recv sock buf))
+         (if (= len 0)
+           (begin
+             (ssocket-close sock)
+             (values))
+           (begin
+             (ssocket-send sock buf 0 len)
```

Comment:
this should be `ssocket-send-all` to ensure that the entire buffer is send even if a partial write happens.